enhancements/machine-api/converting-machine-api-to-cluster-api.md

```diff
@@ -0,0 +1,825 @@
+
+The users intention to migrate will be persisted, and the sync controller will continue to attempt to migrate the resource until the condition is resolved, but, the user is also informed of the reason for the failure synchronously.
+
+This extension is not required for the minimum viable version of this project, but, will be added in a future iteration, prior to the GA release.
+
+### API Extensions
+
+This enhancement introduces a new `authoritativeAPI` field on `spec` and `status` of Machine API resources.
+Users will set the value of this field to determine which of the Machine API or Cluster API controllers will own the resource.
+
+The `status` will also include a `synchronizedGeneration` field to indicate the generation of the authoritative resource that the non-authoritative resource is synchronised with.
+
+Machine API conditions will also include a new `Synchronized` condition type, which will be used to indicate when the non-authoritative resource is fully synchronised with the authoritative resource.
+
+The `Paused` condition will be added to Machine API and Cluster API (upstream first) to allow controllers to acknowledge the change in the authoritative API.
+
+Admission control will prevent changes to non-authoritative resources that do not come from the sync controller, bar a small number of fields that are allowed to be updated by the user.
+
+#### Authority of resources and controller reconciliations
+
+To ensure that only one controller acts on any resource (Machine, MachineSet etc) at any one time,
+a new field `status.authoritativeAPI` will be implemented.
+The new field will be applied to each Machine API resource and will determine which controller is authoritative for the resource.
+
+The accepted values for the field will be either `machine.openshift.io` or `cluster.x-k8s.io` (the respective API groups for Machine API and Cluster API).
+If the field is not present, it will be added by the sync controller and defaulted to `machine.openshift.io`.
+
+When the Machine API controller attempts to reconcile an object, it must first check for the presence of the field.
+If it is not present, then the Machine API controller should reconcile the object.
+It is expected that the sync controller will default it to `machine.openshift.io` in this case.
+
+In Machine API, checks will be added at the beginning of each reconcile loop so that the controller can determine whether or not to reconcile the resource.
+Where the resource is not authoritative, the controller should set the `Paused` condition to `True` and not perform any further reconciliation.
+
+In Cluster API, the same checks will be performed, but instead by the sync controller.
+The sync controller will leverage Cluster API's built in pause mechanism to prevent the Cluster API controllers from reconciling the resource. 
+
+Machine API controllers will be allowed to reconcile any object that is either:
+
+* Missing the authoritative field in the status (as today)
```

Comment:
Is this safe? Might need to caveat this with, empty in spec too. If it's set in spec but not status, this just means the sync controller hasn't got there yet